src/vision/status_quo/barbara_writes_a_runtime_agnostic_lib.md

```diff
@@ -0,0 +1,125 @@
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from
+real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async
+Rust programmers face today.
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR
+making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories
+[cannot be wrong], only inaccurate). Alternatively, you may wish to
+[add your own status quo story][htvsq]!
+
+## The story
+
+Barbara and Alan work at AmoolgeSoft, where many teams are switching from Java to Rust. These teams
+have many different use cases and various adoption stories. Some teams are happy users of tokio,
+others happy users of async-std, and others still are using custom runtimes for highly specialized
+use cases.
+
+Barbara is tasked with writing a library for a custom protocol, SLOW (only in use at AmoogleSoft)
+and enlists the help of Alan in doing so. Alan is already aware that [not all libraries in Rust work
+with all runtimes][nalirwwar]. Alan and Barbara start by writing a parser which works on
+`std::io::Read` and get their tests working with `String`s. After this they contemplate the question
+of how to accept a TCP connection.
+
+### Incompatible `AsyncRead` traits
+
+Alan asks Barbara what is the async equivalent is of `std::io::Read`, and Barbara sighs and says
+that there isn't one. Barbara brings up tokio's and the [futures crate]'s versions of `AsyncRead`.
+Barbara decides not to talk about `AsyncBufRead` for now.
+
+Barbara and Alan decide to use the future's `AsyncRead` for no other reason other than it is
+runtime-agnostic. Barbara tells Alan not to worry as they can translate between the two. With some
+effort (**TODO:** maybe link a story about `Pin`? which one?) they convert their parser to using
+`AsyncRead`.
+
+Alan, excited about the progress they've made, starts working on hooking this up to actual TCP
+streams. Alan looks at async-std and tokio and notices their interfaces for TCP are quite different.
+Alan waits for Barbara to save the day.
+
+Barbara helps abstract over TCP listener and TCP stream (**TODO:** code example). One big hurdle is
+that tokio uses `AsyncRead` from their own crate and not the one from `futures` crate.
+
+### Task spawning
+
+After getting the TCP handling part working, they now want to spawn tasks for handling each incoming
+TCP connection. Again, to their disappointment, they find that there's no runtime-agnostic way to do
+that.
+
+Unsure on how to do this, they do some searching and find the [`agnostik`] crate. They reject it
+because this only supports N number of runtimes and their custom runtime is not one of them.
+However it gives them the idea to provide a trait for specifying how to spawn tasks on the runtime.
+Barbara points out that this has disadvantage of working against orphan rules meaning that either
```

Comment:
We might want to either expand on this here or perhaps better provide a footnote explaining this. There's also now an issue tracking this: #180

Reply:
I think a link to this issue would suffice here but feel free to suggest what further info should be provided here.